### PR TITLE
[Feat] V10 Installation & Upgrading Script - Major Update

### PR DESCRIPTION
**Pull request by KevinLu2000**
1. Support upgrading from ***V9.x*** to ***V10*** (Or ***V10.x*** to ***latest***)
2. ~~Automatically create a full backup before upgrading.~~
3. Support both ***en_us*** and ***zh_cn*** locale
4. Add option to install ***daemon***, ***web***, or ***both***. (-c ***web/daemon/all***). If not specified, default is ***all***.)
5. By default, the service will be installed as user "***mcsm***"
6. Add option to run the service as ***root*** (-u root)
7. For ***en_us*** version, use GitHub as the download source. 
8. Optimize display styles.
9. Revise some texts and details

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ echo_cyan() {
 echo_red() {
     printf '\033[1;31m%b\033[0m\n' "$@"
 }
-
+vv
```
Comment (unitwk) [inline setup.sh:22]: 这个 vv 是不是不小心多写了

Comment (KevinLu2000) [reply setup.sh]: 没错, 复制原始代码的时候可能不小心打多了，已修复。

**Comment by KevinLu2000** [comment]
2024-05-06: Backup function has been disabled due to potential large file size being archived.

**Comment by KevinLu2000** [comment]
2024-05-09: Update function signatures.

**Pull request merged**